src/t_string.c

```diff
@@ -89,14 +98,50 @@ void setGenericCommand(client *c, int flags, robj *key, robj **valref, robj *exp
     found = (lookupKeyWriteWithLink(c->db,key,&link) != NULL);
 
     if ((flags & OBJ_SET_NX && found) ||
-        (flags & OBJ_SET_XX && !found))
+        (flags & (OBJ_SET_XX | OBJ_SET_IFEQ | OBJ_SET_IFDEQ) && !found))
     {
         if (!(flags & OBJ_SET_GET)) {
             addReply(c, abort_reply ? abort_reply : shared.null[c->resp]);
         }
         return;
     }
 
+    /* Handle conditional set operations - only set if key is found and condition
+     * is met - otherwise return nil. */
+    if (found && (flags & (OBJ_SET_IFEQ | OBJ_SET_IFNE | OBJ_SET_IFDEQ | OBJ_SET_IFDNE))) {
+        kvobj *current = lookupKeyRead(c->db, key);
+        if (checkType(c, current, OBJ_STRING)) {
+            return;
+        }
+
+        if (flags & OBJ_SET_IFEQ || flags & OBJ_SET_IFNE) {
+            robj *current_decoded = getDecodedObject(current);
+            int condition = (flags & OBJ_SET_IFEQ) ?
+                            sdscmp(current_decoded->ptr, match_value->ptr) == 0 :
+                            sdscmp(current_decoded->ptr, match_value->ptr) != 0;
+            if (!condition) {
+                decrRefCount(current_decoded);
+                if (!(flags & OBJ_SET_GET)) {
+                    addReply(c, abort_reply ? abort_reply : shared.null[c->resp]);
```

Comment:
I'm not sure if this design is intentional. If there is a GET parameter and the setting fails, will we still return the current value? i think regardless of whether there is a GET parameter or not, as long as the setting fails, should it return a failure?